updateParamGroups.iLogicVb

```diff
@@ -2,123 +2,245 @@
+			Exit Sub
+		End If
+
+		If DoesGroupExist(NewGroupName) = false Then
+''			msgbox("DoesGroupExist = false, addingGroup...")
+			AddGroup(NewGroupName)
+		End If
+	
+		NewGroup = GetGroupByName(NewGroupName)
+	
+		If DoesGroupContainParameter(NewName) = false Then
+''			msgbox("DoesGroupContainParameter = false, adding parameter to group...")
+			AddParameterToGroup(NewName, NewGroup)
+			msgBox("Added parameter " & NewName & " to " & NewGroupName)
+		End If
+
+		oDoc.Update()
+
 	End Sub
 
-	Private Sub getDocumentMembers()
-		doc = ThisDoc.Document
-		Select Case doc.DocumentType
-		Case kPartDocumentObject, kAssemblyDocumentObject
-			oParams = doc.ComponentDefinition.Parameters
-		Case kDrawingDocumentObject
-			oParams = doc.Parameters
-		Case Else
-			Throw New NotImplementedException
-		End Select
-		paramGroups = oParams.CustomParameterGroups
+
+'### Utility Functions ###'
+
+	Private Sub AddGroup(groupName as String)
+		Dim group as CustomParameterGroup
+			group = paramGroups.Add(groupName, groupName)
 	End Sub
 
-	Private Sub removeParamFromGroup(name as String, group as CustomParameterGroup)
+	Private Sub AddParameterToGroup(name as String, group as CustomParameterGroup)
 		Try
-			Dim paramList As IEnumerator = group.GetEnumerator()
-			Dim param As Parameter
-			For i As Integer = 0 To group.Count Step 1
-				paramList.MoveNext()
-				If paramList.Current.Name = name
-					group.Remove(i + 1)
-					Exit For
-				End If
-			Next
+			Dim oParam As Inventor.Parameter = oParameters(name)
+			group.Add(oParam)
 		Catch ex As ArgumentException
 		End Try
 	End Sub
 
-	Private Sub deleteEmptyGroup(group as CustomParameterGroup)
+	Private Sub DeleteGroup(group as CustomParameterGroup)
 		Try
-			If Not group.Count > 0
-				group.Delete()
-			End If
+			group.Delete()
 		Catch ex As ArgumentException
 		End Try
 	End Sub
 
-	Private Function addGroup(groupName as String)
-		Dim group as CustomParameterGroup
-		Try
-			group = paramGroups.Item(groupName)
-		Catch ex As ArgumentException
-			group = paramGroups.Add(groupName, groupName)
-		End Try
-		Return group
+	Private Function DoesGroupExist(value As String) As Boolean
+		dim group As Inventor.CustomParameterGroup
+
+		for each group in paramGroups
+			if group.DisplayName = value Then
+				return true
+			end If
+		next
+
+		return false
 	End Function
 
-	Private Sub addParamToGroup(name as String, group as CustomParameterGroup)
-		Try
-			Dim oParam As Parameter = oParams(name)
-			group.Add(oParam)
-		Catch ex As ArgumentException
-		End Try
+	Private Function DoesGroupContainParameter(value As String) As Boolean
+		dim tempGroup As Inventor.CustomParameterGroup
+		dim tempParameter As Inventor.Parameter
+		for each tempGroup in paramGroups
+			for each tempParameter in tempGroup
+				if tempParameter.Name = value Then
+					return true
+				end If
+			next
+		next
+
+		return false
+	End Function
+
+	Private Function DoesGroupNameExist(value as String) As Boolean
```

Comment:
I think you already have a function for this, `StringContainsGroupName(value as String)`